src/actions/segment/segment.ts

```diff
@@ -28,6 +44,61 @@ export class SegmentAction extends Hub.Action {
   supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
   requiredFields = [{ any_tag: this.allowedTags }]
 
+  taggedFields(fields: any, tags: string[]) {
```

Comment:
you added tests for these (good!) but if you didn't need to you could mark these as `protected`